tariff/tariffs.go

```diff
@@ -25,6 +25,55 @@ func NewTariffs(currency currency.Unit, grid, feedin, planner api.Tariff) *Tarif
 	}
 }
 
+func currentPrice(t api.Tariff) (float64, error) {
+	if t != nil {
+		if rr, err := t.Rates(); err == nil {
+			if r, err := rr.Current(time.Now()); err == nil {
+				return r.Price, nil
+			}
+		}
+	}
+	return 0, api.ErrNotAvailable
+}
+
+// CurrentGridPrice returns the current feedin price.
+func (t *Tariffs) CurrentGridPrice() (float64, error) {
+	return currentPrice(t.Grid)
+}
+
+// CurrentFeedInPrice returns the current feedin price.
+func (t *Tariffs) CurrentFeedInPrice() (float64, error) {
+	return currentPrice(t.FeedIn)
+}
+
+// CurrentEffectivePrice calculates the real energy price based on self-produced and grid-imported energy.
+func (t *Tariffs) CurrentEffectivePrice(greenShare float64) (float64, error) {
```

Comment:
Das gehört m.E. nicht in die Tariffs. Greenshare ist ein (hier) "fremdes" Konzept.

Reply:
Done. Effective* sind zum GreenShare in die Site verschoben.